packages/now-routing-utils/test/superstatic.spec.js

```diff
@@ -0,0 +1,267 @@
+    'path/to/file.txt',
+    'file.js',
+    'path/to/file.js',
+    'file.html',
+    'path/to/file.html',
+  ]);
+  const expected = [
+    {
+      src: 'file.html',
+      headers: { Location: 'file' },
+      status: 307,
+      continue: true,
+    },
+    {
+      src: 'path/to/file.html',
+      headers: { Location: 'path/to/file' },
+      status: 307,
+      continue: true,
+    },
+    { src: 'file', dest: 'file.html', continue: true },
+    { src: 'path/to/file', dest: 'path/to/file.html', continue: true },
+  ];
+  deepEqual(actual, expected);
+
+  const mustMatch = [
+    ['file.html'],
+    ['path/to/file.html'],
+    ['file'],
+    ['path/to/file'],
+  ];
+
+  const mustNotMatch = [
+    ['file2.html', 'afile.html'],
+    ['path/to/file2.html', 'path/to/file'],
+    ['file2', 'afile'],
+    ['path/to/file2', 'file'],
+  ];
+
+  assertRegexMatches(actual, mustMatch, mustNotMatch);
+});
+
+test('convertRedirects', () => {
+  const actual = convertRedirects([
+    { source: '/some/old/path', destination: '/some/new/path' },
+    {
+      source: '/firebase/(.*)',
+      destination: 'https://www.firebase.com',
+      statusCode: 302,
+    },
+    {
+      source: '/projects/:id/:action',
+      destination: '/projects.html?id=:id&action=:action',
+    },
+    { source: '/old/:segment/path', destination: '/new/path/:segment' },
+  ]);
+
+  const expected = [
+    {
+      src: '^\\/some\\/old\\/path$',
+      headers: { Location: '/some/new/path' },
+      status: 307,
+      continue: true,
+    },
+    {
+      src: '^\\/firebase\\/(.*)$',
+      headers: { Location: 'https://www.firebase.com' },
+      status: 302,
+      continue: true,
+    },
+    {
+      src: '^\\/projects\\/([^\\/]+?)\\/([^\\/]+?)$',
+      headers: { Location: '/projects.html?id=$1&action=$2' },
+      status: 307,
+      continue: true,
+    },
+    {
+      src: '^\\/old\\/([^\\/]+?)\\/path$',
+      headers: { Location: '/new/path/$1' },
+      status: 307,
+      continue: true,
+    },
+  ];
+
+  deepEqual(actual, expected);
+
+  const mustMatch = [
+    ['/some/old/path'],
+    ['/firebase/one', '/firebase/2', '/firebase/-', '/firebase/dir/sub'],
+    ['/projects/one/edit', '/projects/two/edit'],
+    ['/old/one/path', '/old/two/path'],
+  ];
+
+  const mustNotMatch = [
+    ['/nope'],
+    ['/fire', '/firebasejumper/two'],
+    ['/projects/edit', '/projects/two/three/delete', '/projects'],
+    ['/old/path', '/old/two/foo', '/old'],
+  ];
+
+  assertRegexMatches(actual, mustMatch, mustNotMatch);
+});
+
+test('convertRewrites', () => {
+  const actual = convertRewrites([
+    { source: '/some/old/path', destination: '/some/new/path' },
+    { source: '/firebase/(.*)', destination: 'https://www.firebase.com' },
+    { source: '/projects/:id/edit', destination: '/projects.html?id=$1' },
+  ]);
+
+  const expected = [
+    { src: '^\\/some\\/old\\/path$', dest: '/some/new/path', continue: true },
+    {
+      src: '^\\/firebase\\/(.*)$',
```

Comment:
We don't need to escape slashes anywhere. The only place it makes sense is in a regex literal (`/patt\\/path/`)

Reply:
Unfortunately, this is the behavior from `path-to-regexp` which converts the segments to regex. Will this cause problems?

Reply:
Can we normalize this too? I imagine a UI for seeing the generated routes could be helpful and the escaping makes it look more complicated than it really is

Reply:
I don't think users are going to be seeing the generated `routes`. There is no reason to because all the new keys cover their use case.

Basically the `routes` key is going to become like `builds` key and only be used by our team internally.

Reply:
I think for debugging purposes, it's very useful

Reply:
Could we add this to the normalization :pray: It cuts down on stored bytes in the database too